staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -532,9 +532,29 @@ func (r *Request) Watch() (watch.Interface, error) {
 		}
 		return nil, fmt.Errorf("for request '%+v', got status: %v", url, resp.StatusCode)
 	}
-	framer := r.serializers.Framer.NewFrameReader(resp.Body)
-	decoder := streaming.NewDecoder(framer, r.serializers.StreamingSerializer)
-	return watch.NewStreamWatcher(restclientwatch.NewDecoder(decoder, r.serializers.Decoder)), nil
+
+	framer := r.serializers.Framer
+	streamingSerializer := r.serializers.StreamingSerializer
+	normalDecoder := r.serializers.Decoder
+
+	// check if we need to renegotiate for different content types (like done in
+	// transformResponse for non-streaming requests)
+	contentType := resp.Header.Get("Content-Type")
+	if len(contentType) > 0 && len(r.content.ContentType) > 0 && contentType != r.content.ContentType {
+		mediaType, params, err := mime.ParseMediaType(contentType)
+		if err != nil {
+			return nil, err
+		}
+		normalDecoder, streamingSerializer, framer, err = r.serializers.RenegotiatedStreaming(mediaType, params)
```

Comment:
this assumes RenegotiatedStreaming is non-nil?

Reply:
the code seemed to assume StreamingSerializer was non-nil before.  I'll add a check in to be safe.